src/fsipy/automatedPreprocessing/automated_preprocessing.py

```diff
@@ -372,40 +381,45 @@ def run_pre_processing(input_model, verbose_print, smoothing_method, smoothing_f
                 distance_to_sphere = distance_to_spheres_solid_thickness(distance_to_sphere,
                                                                          file_name_distance_to_sphere_solid_thickness)
             else:
-                print("ERROR: Invalid parameters for variable solid thickness. This should be "+
+                print("ERROR: Invalid parameters for variable solid thickness. This should be " +
                       "given as four parameters: 'offset', 'scale', 'min' and 'max.")
                 sys.exit(-1)
         else:
             distance_to_sphere = read_polydata(file_name_distance_to_sphere_solid_thickness)
-                
     else:
-        pass  # FIXME: how to set constant thickness
+        if len(solid_thickness_parameters) != 1 or solid_thickness_parameters[0] <= 0:
+            print("ERROR: Invalid parameter for constant solid thickness. This should be a " +
+                  "single number greater than zero.")
+            sys.exit(-1)
             
     # Compute mesh
     if not path.isfile(file_name_vtu_mesh):
         print("--- Computing mesh\n")
         try:
-            mesh, remeshed_surface = generate_mesh(distance_to_sphere, add_boundary_layer, meshing_method,
-                                                   solid_thickness, solid_thickness_parameters)
+            mesh, remeshed_surface = generate_mesh(distance_to_sphere, solid_thickness, solid_thickness_parameters)
         except Exception:
             distance_to_sphere = mesh_alternative(distance_to_sphere)
-            mesh, remeshed_surface = generate_mesh(distance_to_sphere, add_boundary_layer, meshing_method,
-                                                   solid_thickness, solid_thickness_parameters)
+            mesh, remeshed_surface = generate_mesh(distance_to_sphere, solid_thickness, solid_thickness_parameters)
 
         assert mesh.GetNumberOfPoints() > 0, "No points in mesh, try to remesh."
         assert remeshed_surface.GetNumberOfPoints() > 0, "No points in surface mesh, try to remesh."
 
         if mesh.GetNumberOfPoints() < remeshed_surface.GetNumberOfPoints():
             print("--- An error occurred during meshing. Will attempt to re-mesh \n")
-            mesh, remeshed_surface = generate_mesh(distance_to_sphere, add_boundary_layer, meshing_method,
-                                                   solid_thickness, solid_thickness_parameters)
+            mesh, remeshed_surface = generate_mesh(distance_to_sphere, solid_thickness, solid_thickness_parameters)
 
         write_mesh(compress_mesh, file_name_surface_name, file_name_vtu_mesh, file_name_xml_mesh,
                    mesh, remeshed_surface)
 
+        # Add .gz to XML mesh file if compressed
+        if compress_mesh:
+            file_name_xml_mesh = file_name_xml_mesh + ".gz"
```

Comment:
I think we should use `pathlib` for filenames in general, then one can use `path.with_suffix(".gz")` to change the file type.

Reply:
Yes, that would be good.